pkg/simulation/run.go

```diff
@@ -133,6 +133,12 @@ func queuePhase(s *Simulation) (stateFn, error) {
 		// we do the same skip here as if eval doesn't have any more ations
 		return s.advanceFrames(1, queuePhase)
 	}
+	// check if the next queue item is valid
+	// example: most sword characters can't do charge if the previous action was not attack
+	currentChar := s.C.Player.ActiveChar()
+	if err := currentChar.NextQueueItemIsValid(*next); err != nil {
```

Comment:
this check here should be just right after the swap check

Reply:
also probably ok to pass next *action.Eval? not sure if any benefit to having signature non pointer and having to dereference here